source/Phage.h

```diff
@@ -281,43 +358,20 @@ class Phage: public Symbiont {
   void Process(size_t location) {
     if(lysis_enabled && !GetHost().IsNull()) { //lysis enabled and phage is in a host
       if(!lysogeny){ //phage has chosen lysis
-        if(GetBurstTimer() >= burst_time ) { //time to lyse!
-          emp::vector<emp::Ptr<Organism>>& repro_syms = my_host->GetReproSymbionts();
- 
-          //Record the burst size and count
-          emp::DataMonitor<double>& data_node_burst_size = my_world->GetBurstSizeDataNode();
-          data_node_burst_size.AddDatum(repro_syms.size());
-          emp::DataMonitor<int>& data_node_burst_count = my_world->GetBurstCountDataNode();
-          data_node_burst_count.AddDatum(1);
- 
-          for(size_t r=0; r<repro_syms.size(); r++) {
-            my_world->SymDoBirth(repro_syms[r], location);
-          }
-          my_host->ClearReproSyms();
-          my_host->SetDead();
-          return;
- 
-        } else { //not time to lyse
-          IncBurstTimer();
-          if(sym_lysis_res == 0) {
-            std::cout << "Lysis with a sym_lysis_res of 0 leads to an \
-            infinite loop, please change" << std::endl;
-            std::exit(1);
-          }
-          while(GetPoints() >= sym_lysis_res) {
-            emp::Ptr<Organism> sym_baby = reproduce();
-            my_host->AddReproSym(sym_baby);
-            SetPoints(GetPoints() - sym_lysis_res);
-          }
-        }
+        HorizontalTransmission(location);
       }
       else if(lysogeny){ //phage has chosen lysogeny
-        //check if the phage's host should become susceptible again
-        if(random->GetDouble(0.0, 1.0) <= my_config->PROPHAGE_LOSS_RATE()){
+        double rand_chance = random->GetDouble(0.0, 1.0);
+        if (rand_chance <= induction_chance){//phage has chosen to induce and perform lysis
+          lysogeny = false;
+          HorizontalTransmission(location);
```

Comment:
Does it work to not call HorizontalTransmission immediately and instead just set lysogeny to false and wait until the next update to actually start the lysis process?